packages/components/dialog/src/use-dialog.ts

```diff
@@ -25,9 +26,70 @@ export const useDialog = (
   const normalizeWidth = computed(() =>
     isNumber(props.width) ? `${props.width}px` : props.width
   )
+  const transform = ref({
+    offsetX: 0,
+    offsetY: 0,
+  })
+  function handleMouseDown(e: MouseEvent) {
+    if (
+      !e.target.classList.contains('el-dialog__header') &&
+      !e.target.classList.contains('el-dialog__body') &&
+      !e.target.classList.contains('el-dialog__footer')
+    ) {
+      return
+    }
+    if (props.fullscreen || !props.draggable) return
+    if (e.button !== 0 || !wrapper.value) return
+    const { offsetX, offsetY } = transform.value
+    const startX = e.clientX
+    const startY = e.clientY
+
+    const divLeft = wrapper.value.clientLeft
+    const divRight = wrapper.value.clientLeft + wrapper.value.clientWidth
+    const divTop = wrapper.value.clientTop
+    const divBottom = wrapper.value.clientTop + wrapper.value.clientHeight
+
+    const dragHandler = rafThrottle((ev: MouseEvent) => {
+      transform.value = {
+        offsetX: offsetX + ev.clientX - startX,
+        offsetY: offsetY + ev.clientY - startY,
+      }
+    })
+    const removeMousemove = useEventListener(document, 'mousemove', dragHandler)
+    useEventListener(document, 'mouseup', (evt) => {
+      if (props.resettable) {
+        const mouseX = evt.clientX
+        const mouseY = evt.clientY
+        if (
+          mouseX < divLeft ||
+          mouseX > divRight ||
+          mouseY < divTop ||
+          mouseY > divBottom
+        ) {
+          reset()
+        }
+      }
+      removeMousemove()
+    })
+
+    e.preventDefault()
+  }
+
+  function reset() {
+    transform.value = {
+      offsetX: 0,
+      offsetY: 0,
+    }
+  }
 
   const style = computed<CSSProperties>(() => {
-    const style: CSSProperties = {}
+    let style: CSSProperties = {}
+    if (wrapper && props.draggable) {
+      style = {
+        left: `${transform.value.offsetX}px`,
```

Comment:
consider using translate3d instead left and top

Reply:
fine